auth/v1.js

```diff
@@ -0,0 +1,615 @@
+  name:       'client',
+  input:      undefined,
+  stability:  'stable',
+  output:     'get-client-response.json#',
+  title:      "Words...",
+  description: [
+    "Words..."
+  ].join('\n')
+}, async function(req, res) {
+  let clientId = req.params.clientId;
+
+  // Load client
+  let client = await this.Client.load({clientId}, true);
+
+  if (!client) {
+    return res.status(404).json({message: "Client not found!"});
+  }
+
+  res.reply(client.json());
+});
+
+
+/** Create client */
+api.declare({
+  method:     'put',
+  route:      '/clients/:clientId',
+  name:       'createClient',
+  input:      'create-client-request.json#',
+  output:     'create-client-response.json#',
+  scopes:     [['auth:create-client:<clientId>']],
+  deferAuth:  true,
+  stability:  'stable',
+  title:      "Create Client",
+  description: [
+    "Words..."
+  ].join('\n')
+}, async function(req, res) {
+  let clientId  = req.params.clientId;
+  let input     = req.body;
+
+  // Check scopes
+  if (!req.satisfies({clientId})) {
+    return;
+  }
+
+  var accessToken = slugid.v4() + slugid.v4();
+  let client = await this.Client.create({
+    clientId:     clientId,
+    description:  input.description,
+    accessToken:  accessToken,
+    expires:      new Date(input.expires),
+    details: {
+      created:      new Date().toJSON(),
+      lastModified: new Date().toJSON(),
+      lastDateUsed: new Date().toJSON(),
+      lastRotated:  new Date().toJSON()
+    }
+  }).catch(async (err) => {
+    // Only handle
+    if (err.code !== 'EntityAlreadyExists') {
+      throw err;
+    }
+
+    // Load client
+    let client = await this.Client.load({clientId});
+
+    // If stored client different or older than 15 min we return 409
+    let created = new Date(client.details.created).getTime();
+    if (client.description !== input.description ||
+        client.expires.getTime() !== new Date(input.expires).getTime() ||
+        created > Date.now() - 15 * 60 * 1000) {
+      res.status(409).json({
+        message: "client with same clientId already exists, possibly" +
+                 "an issue with retry logic or idempotency"
+      });
+      return null;
+    }
+
+    return client;
+  });
+
+  // If no client it was already created
+  if (!client) {
+    return;
+  }
+
+  // Send pulse message
+  await Promise.all([
+    this.publisher.clientCreated({clientId}),
+    this.resolver.reloadClient(clientId)
+  ]);
+
+  // Create result with access token
+  let result = client.json();
+  result.accessToken = client.accessToken;
+  return res.reply(result);
+});
+
+
+/** Reset access token for client */
+api.declare({
+  method:     'post',
+  route:      '/clients/:clientId/reset',
+  name:       'resetAccessToken',
+  input:      undefined,
+  output:     'create-client-response.json#',
+  scopes:     [['auth:reset-access-token:<clientId>']],
+  deferAuth:  true,
+  stability:  'stable',
+  title:      "Reset `accessToken`",
+  description: [
+    "Words..."
```

Comment:
Unless I misunderstand, this will immediately invalidate all temporary credentials issued with this clientId, too, right?  Worth noting in the docs.